drivers/soundwire/generic_bandwidth_allocation.c

```diff
@@ -375,11 +413,60 @@ static int sdw_compute_bus_params(struct sdw_bus *bus)
 				(bus->params.max_dr_freq >>  clk_buf[i]) :
 				clk_buf[i] * SDW_DOUBLE_RATE_FACTOR;
 
-		if (curr_dr_freq * (mstr_prop->default_col - 1) <
+		if (curr_dr_freq * (mstr_prop->default_col - 1) >=
 		    bus->params.bandwidth * mstr_prop->default_col)
-			continue;
+			break;
 
-		break;
+		list_for_each_entry(m_rt, &bus->m_rt_list, bus_node) {
+			/*
+			 * Get the first s_rt that will be used to find the available lane that
+			 * can be used.
+			 */
+			s_rt = list_first_entry(&m_rt->slave_rt_list,
+						struct sdw_slave_runtime, m_rt_node);
+			slave_prop = &s_rt->slave->prop;
+
+			/*
+			 * Find available manager lanes that connected to the first Peripheral.
+			 * No need to check all Peripherals available lanes because we can't use
+			 * multi-lane if we can't find any available lane for the first Peripheral.
+			 */
+			for (l = 1; l < SDW_MAX_LANES; l++) {
+				if (!slave_prop->lane_maps[l])
+					continue;
+
+				dev_dbg(bus->dev, "%s: trying lane %d\n", __func__, l);
+				required_bandwidth = 0;
+				list_for_each_entry(m_p_rt, &m_rt->port_list, port_node) {
+					required_bandwidth += m_rt->stream->params.rate *
+						hweight32(m_p_rt->ch_mask) *
+						m_rt->stream->params.bps;
+				}
+				if (required_bandwidth <=
+				    curr_dr_freq - bus->lane_used_bandwidth[l]) {
+					/* Check if m_lane is connected to all Peripherals */
+					if (!check_all_peripherals_connected(m_rt,
+						slave_prop->lane_maps[l])) {
+						dev_dbg(bus->dev,
+							"some Peripherals are not connected to %d\n",
+							slave_prop->lane_maps[l]);
```

Comment:
what does this means?
Some is connected, some is not?

Confusing.

Reply:
Both Managers and Peripherals can have multiple lanes. They could be connected by any lane. For example, Peripheral A could connect to the Manager with lane 0, 1 and 2. And Peripheral A could connect to the Manager with lane 0, 1 and 3. In that case, we can only use lane 1 as the second lane.